src/plugins/Sharesite/Inserter.java

```diff
@@ -31,27 +34,47 @@ public Inserter() {
 	@Override
 	public void run() {
 		Freesite nextToInsert;
-
+		
 		while (true) {
+			Integer currentHour = Calendar.getInstance(TimeZone.getTimeZone("UTC"), Locale.US)
+				.get(Calendar.HOUR_OF_DAY); // 0-23
+			Integer currentMinute = Calendar.getInstance(TimeZone.getTimeZone("UTC"), Locale.US)
+				.get(Calendar.MINUTE); // 0-59
 			// Quick do everything that requires locking
-			synchronized (this) {
-				nextToInsert = null;
-
-				if (!queuedInserts.isEmpty()) {
-					nextToInsert = queuedInserts.removeFirst();
-				}
-			}
-
+			nextToInsert = getNextToInsert(currentHour, queuedInserts);
+			
 			// Now safely perform the blocking inserts
 			if (nextToInsert != null) {
+				// TODO: Wait for a random fraction of the remaining
+				// part of the current hour to prevent detection of
+				// people who click insert during the insertHour.
+				if (!nextToInsert.getInsertHour().equals(-1)) { // no instant insert
```

Comment:
`Integer`s don’t have to be compared with `.equals()`, the VM automatically boxes/unboxes as required.